src/metabase/models/card.clj

```diff
@@ -1004,29 +1004,28 @@
            (progress? display))
        (< 1 (count (get-in new-card [:dataset_query :query :breakout])))))
 
-(defn- delete-alert-and-notify!
+(defn delete-alert-and-notify!
   "Removes all of the alerts and notifies all of the email recipients of the alerts change."
-  [topic actor alerts]
-  (t2/delete! :model/Pulse :id [:in (mapv u/the-id alerts)])
-  (events/publish-event! topic {:alerts alerts, :actor actor}))
+  [topic actor card]
+  (when-let [card-notifications (seq (models.notification/notifications-for-card (:id card)))]
+    (t2/delete! :model/Notification :id [:in (map :id card-notifications)])
+    (events/publish-event! topic {:card          card
+                                  :actor         actor
+                                  :notifications card-notifications})))
 
 (defn- delete-alerts-if-needed! [& {:keys [old-card new-card actor]}]
-  ;; If there are alerts, we need to check to ensure the card change doesn't invalidate the alert
-  (when-let [alerts (binding [models.pulse/*allow-hydrate-archived-cards* true]
-                      (not-empty (models.pulse/retrieve-alerts-for-cards {:card-ids [(u/the-id new-card)]})))]
-    (cond
+  (cond
+    (card-archived? old-card new-card)
+    (delete-alert-and-notify! :event/card-update.notification-deleted.card-archived actor new-card)
 
-      (card-archived? old-card new-card)
-      (delete-alert-and-notify! :event/card-update.alerts-deleted.card-archived actor alerts)
+    (or (display-change-broke-alert? old-card new-card)
+        (goal-missing? old-card new-card)
+        (multiple-breakouts? new-card))
+    (delete-alert-and-notify! :event/card-update.notification-deleted.card-became-invalid actor new-card)
```

Comment:
Ohhh we can actually pass the reason here! Encoding the reason in the topic name feels a bit weird to me, but if it's the current approach 🤷‍♂️